EssentialFeed/EssentialFeed/Image Comments Presentation/ImageCommentsPresenter.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  ImageCommentsPresenter.swift
+//  EssentialFeed
+//
+//  Created by Bogdan Poplauschi on 28/03/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+
+public protocol ImageCommentsView {
+	func display(_ viewModel: ImageCommentsViewModel)
+}
+
+public protocol ImageCommentsLoadingView {
+	func display(_ viewModel: ImageCommentsLoadingViewModel)
+}
+
+public protocol ImageCommentsErrorView {
+	func display(_ viewModel: ImageCommentsErrorViewModel)
+}
+
+public final class ImageCommentsPresenter {
+	private let imageCommentsView: ImageCommentsView
+	private let loadingView: ImageCommentsLoadingView
+	private let errorView: ImageCommentsErrorView
+	private let currentDate: Date
+	
+	public static var title: String {
+		NSLocalizedString(
+			"IMAGE_COMMENTS_VIEW_TITLE",
+			tableName: "ImageComments",
+			bundle: Bundle(for: ImageCommentsPresenter.self),
+			comment: "Title for the image comments view")
+	}
+	
+	private var errorMessage: String {
+		NSLocalizedString(
+			"IMAGE_COMMENTS_VIEW_CONNECTION_ERROR",
+			tableName: "ImageComments",
+			bundle: Bundle(for: ImageCommentsPresenter.self),
+			comment: "Error message when loading comments fails"
+		)
+	}
+	
+	public init(
+		imageCommentsView: ImageCommentsView,
+		loadingView: ImageCommentsLoadingView,
+		errorView: ImageCommentsErrorView,
+		currentDate: Date = Date()
```

Comment:
The `currentDate` is supposed to represent the actual current date.

But if you inject an instance of `Date`, it'll be outdated immediately. It won't represent the current date as time passes by.

Instead, you need to inject a component that can *produce* the current date for *now* - every time it should produce a new date representing the `current date`.

It could be a closure, for example:

```suggestion
		currentDate: () -> Date = Date.init
```